broker/broker.go

```diff
@@ -223,40 +221,64 @@ func (b *Broker) Bind(context context.Context, instanceID, bindingID string, det
 		return binding, err
 	}
 
-	session, err := MongoConnection(deployment.Connection.Direct[0], deployment.CACertificateBase64, b.RequireTLS)
-	if err != nil {
-		return binding, err
-	}
-	err = session.DB(dbName).UpsertUser(&mgo.User{
-		Username: username,
-		Password: password,
-		Roles:    []mgo.Role{mgo.RoleReadWrite},
-	})
-	if err != nil {
-		return binding, err
-	}
-
-	bindingURL, err := url.Parse(deployment.Connection.Direct[0])
-	if err != nil {
-		return binding, err
-	}
-	bindingURL.User = url.UserPassword(username, password)
-	bindingURL.Path = fmt.Sprintf("/%s", dbName)
-
-	// FIXME: Follow up story should fix mongo connection string handling.
-	// Right now we are hardcoding first host from the comma delimited list that Compose provides.
-	// url.Parse() parses mongo connection string wrongly and doesn't return an error
-	// so url.Port() returns port like "18899,aws-eu-west-1-portal.7.dblayer.com:18899"
-	port := strings.Split(bindingURL.Port(), ",")
-
-	binding.Credentials = Credentials{
-		Host:                bindingURL.Hostname(),
-		Port:                port[0],
-		Name:                dbName,
-		Username:            username,
-		Password:            password,
-		URI:                 bindingURL.String(),
-		CACertificateBase64: deployment.CACertificateBase64,
+	// Different database types have different needs for setting up credentials
+	switch deploymentMeta.Type {
+	case "mongodb":
+		session, err := MongoConnection(deployment.Connection.Direct[0], deployment.CACertificateBase64)
+		if err != nil {
+			return binding, err
+		}
+		err = session.DB(dbName).UpsertUser(&mgo.User{
+			Username: username,
+			Password: password,
+			Roles:    []mgo.Role{mgo.RoleReadWrite},
+		})
+		if err != nil {
+			return binding, err
+		}
+		bindingURL, err := url.Parse(deployment.Connection.Direct[0])
+		if err != nil {
+			return binding, err
+		}
+		bindingURL.User = url.UserPassword(username, password)
+		bindingURL.Path = fmt.Sprintf("/%s", dbName)
+		// FIXME: Follow up story should fix mongo connection string handling.
+		// Right now we are hardcoding first host from the comma delimited list that Compose provides.
+		// url.Parse() parses mongo connection string wrongly and doesn't return an error
+		// so url.Port() returns port like "18899,aws-eu-west-1-portal.7.dblayer.com:18899"
+		port := strings.Split(bindingURL.Port(), ",")
+
+		binding.Credentials = Credentials{
+			Host:                bindingURL.Hostname(),
+			Port:                port[0],
+			Name:                dbName,
+			Username:            username,
+			Password:            password,
+			URI:                 bindingURL.String(),
+			CACertificateBase64: deployment.CACertificateBase64,
+		}
+	default:
+		// THINK: maybe this case should be an error?
```

Comment:
I agree too. And will make this an error now.

However, by making that case an error a "fakedb" type will need to be added to this case to allow unit tests something to tests against.

I actually think the hard coding of strings in this switch is actually a bit of a smell and a better implementation of switching between database types (pluggable `Service` interface based on  catalog maybe?) should be considered when actually implementing the elastic_search et al types.